src/Microsoft.ServiceFabric.Actors.Migration/MigrationOrchestrator.cs

```diff
@@ -217,6 +227,67 @@ public async Task StartMigration(CancellationToken cancellationToken)
             }
         }
 
+        private async Task FetchStringActorIds(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var startSN = long.Parse(await Client.GetStringAsync(this.userSettings.KvsEndpoint + MigrationConstants.GetStartSNEndpoint));
+            var endSN = long.Parse(await Client.GetStringAsync(this.userSettings.KvsEndpoint + MigrationConstants.GetEndSNEndpoint));
+
+            if (endSN - startSN < this.userSettings.ItemsPerEnumeration || this.userSettings.ItemsPerEnumeration == -1)
+            {
+                this.ambiguousActorIdHandler.AddRange(await this.GetStringActorIdsFromKvsAsync(startSN, endSN - startSN, cancellationToken));
+            }
```

Comment:
Instead of enumerating by sequence number, use enumeration based on continuation token. Refer to GetActorAsync or GetRemindersAsync. 
You could use chunkStartLsn and chunkEndLsn to ensure we are not overstepping by checking the version field.